From 971215e035073d48930020f08b26ca3ef9fbc009 Mon Sep 17 00:00:00 2001
From: Andrey Zhizhikin <andrey.zhizhikin@leica-geosystems.com>
Date: Sun, 26 Jan 2020 10:15:40 +0000
Subject: [PATCH] gst-libs/gst: fix meson build in nxp fork

- Add missing IMX header file into istall target;
- Allocator's meta data files are missing in meson.build, bring them back.
- Some video sources are missing, bring them also into the build.

Upstream-Status: Pending

Signed-off-by: Andrey Zhizhikin <andrey.zhizhikin@leica-geosystems.com>
---
 gst-libs/gst/allocators/meson.build | 14 +++++++++++++-
 gst-libs/gst/meson.build            |  6 ++++++
 gst-libs/gst/video/meson.build      | 11 +++++++++++
 3 files changed, 30 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/allocators/meson.build b/gst-libs/gst/allocators/meson.build
index 364baeebf..f82d9c3c6 100644
--- a/gst-libs/gst/allocators/meson.build
+++ b/gst-libs/gst/allocators/meson.build
@@ -5,9 +5,21 @@ gst_allocators_headers = [
   'gstphysmemory.h',
   'gstdmabuf.h',
 ]
+
+imx_gst_allocators_headers = [
+  'gstphymemmeta.h',
+  'gstdmabufmeta.h',
+  'gstallocatorphymem.h',
+]
+gst_allocators_headers += imx_gst_allocators_headers
+
 install_headers(gst_allocators_headers, subdir : 'gstreamer-1.0/gst/allocators/')
 
 gst_allocators_sources = [ 'gstdmabuf.c', 'gstfdmemory.c', 'gstphysmemory.c']
+
+imx_gst_allocators_sources = ['gstdmabufmeta.c', 'gstphymemmeta.c', 'gstallocatorphymem.c']
+gst_allocators_sources += imx_gst_allocators_sources
+
 gstallocators = library('gstallocators-@0@'.format(api_version),
   gst_allocators_sources,
   c_args : gst_plugins_base_args + ['-DBUILDING_GST_ALLOCATORS'],
@@ -16,7 +28,7 @@ gstallocators = library('gstallocators-@0@'.format(api_version),
   soversion : soversion,
   darwin_versions : osxversion,
   install : true,
-  dependencies : [gst_dep],
+  dependencies : [video_dep, gst_dep],
 )
 
 allocators_gen_sources  = []
diff --git a/gst-libs/gst/meson.build b/gst-libs/gst/meson.build
index cd3b5b043..1935f46e5 100644
--- a/gst-libs/gst/meson.build
+++ b/gst-libs/gst/meson.build
@@ -11,3 +11,9 @@ subdir('app')
 subdir('allocators')
 # FIXME: gl deps are automagic
 subdir('gl')
+
+# Install dangling imx header, required by other plugins
+gst_imx_header = [
+    'gstimxcommon.h',
+]
+install_headers(gst_imx_header, subdir : 'gstreamer-1.0/')
diff --git a/gst-libs/gst/video/meson.build b/gst-libs/gst/video/meson.build
index b4dfcdf65..d8531f199 100644
--- a/gst-libs/gst/video/meson.build
+++ b/gst-libs/gst/video/meson.build
@@ -35,6 +35,11 @@ video_sources = [
   'videooverlay.c',
 ]
 
+imx_video_sources = [
+  'gstvideohdr10meta.c',
+]
+video_sources += imx_video_sources
+
 video_headers = [
   'colorbalance.h',
   'colorbalancechannel.h',
@@ -70,6 +75,12 @@ video_headers = [
   'video-overlay-composition.h',
   'video-multiview.h',
 ]
+
+imx_video_headers = [
+  'gstvideohdr10meta.h',
+]
+video_headers += imx_video_headers
+
 install_headers(video_headers, subdir : 'gstreamer-1.0/gst/video/')
 
 video_mkenum_headers = [
-- 
2.17.1

